teps/0007-conditions-beta.md

```diff
@@ -301,51 +303,50 @@ A `Task` branch is made up of dependent `Tasks`, where there are two types of de
 - _Resource dependency_: based on resources needed from parent `Task`, which includes `Results` and `Resources`.
 - _Ordering dependency_: based on `runAfter` which provides sequencing of `Tasks` when there may not be resource dependencies.
 
-To provide more flexibility when `WhenExpressions` evaluate to `False`, we propose adding a field - `whenSkipped` - that:
+To provide more flexibility when `WhenExpressions` evaluate to `False`, we propose adding a field - `scope` - that:
 - is used to specify whether to execute `Tasks` that are ordering-dependent on the skipped guarded `Task`
-- defaults to `skipBranch` and users can set it to `runBranch` to allow execution of the rest of the branch
-- can be specified in `Tasks` guarded with `WhenExpressions` only; there will be a validation error if `whenSkipped` is specified in `Tasks` without `WhenExpressions`
-- can take the values `skipBranch` or `runBranch` only; there will be a validation error if anything else is passed to `whenSkipped`
-- can be specified guarded `Tasks` with ordering dependencies only; there will be a validation error if `whenSkipped` is specified in `Tasks` with resource dependencies
+- defaults to `Branch` and users can set it to `Node` to allow execution of the rest of the branch
+- can take the values `Branch` or `Node` only; there will be a validation error if anything else is passed to `scope`
+- can take the value `Node` in guarded `Tasks` with ordering dependencies only; there will be a validation error if `scope` is set to `Node` in `Tasks` with resource dependencies
```

Comment:
This is not 100% clear to me.

![image](https://user-images.githubusercontent.com/2205608/99071880-e1ece980-25aa-11eb-8158-3cea0508cf30.png)


I would expect:
- branch is always possible
- node is always possible if there are no tasks that depend on the guarded one
- node is not possible if there is a task that depends on resources produced by the guarded one

Is this correct?

Reply:
yes, that is correct